BraveWallet/WalletLoadingButton.swift

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveUI
+
+struct WalletLoadingButton<Title: View>: View {
+  @Binding var isLoading: Bool
+  var action: () -> Void
+  var title: Title
+  
+  @ScaledMetric private var length: CGFloat = 20.0
+  
+  init(
+    isLoading: Binding<Bool>,
+    action: @escaping () -> Void,
+    @ViewBuilder title: () -> Title
+  ) {
+    self._isLoading = isLoading
+    self.action = action
+    self.title = title()
+  }
+  
+  var body: some View {
+    Button {
+      if !isLoading {
+        action()
+      }
+    } label: {
+      ZStack {
+        title
+        Circle()
+          .trim(from: 0, to: 0.7)
+          .stroke(.white, lineWidth: 2)
+          .frame(width: length, height: length)
+          .rotationEffect(.degrees(isLoading ? 360 : 0))
+          .animation(.linear(duration: 0.5).repeatForever(autoreverses: false), value: isLoading)
+          .opacity(isLoading ? 1 : 0)
```

Comment:
Just use `ProgressView` instead?

Reply:
I was trying to make it look the same as Desktop.

Reply:
We'd probably want to use a `ProgressViewStyle` in that case but I'd say for now just replace with `ProgressView` and we can revisit later

Reply:
ok. will update.